### DISCUSSION TITLE
import new object into the project

### DISCUSSION THREAD

**Discussion by brigidatm** [RayTracing/raytracing.github.io#953] [Q&A]
how can i import a new object into my project? for example an object in obj format that I downloaded from the internet

**Comment by hollasch** [comment]
You'd have to program up a way to convert OBJ primitives to spheres and rects. If I recall, OBJ also includes tessellated surfaces, so you'd need to implement triangle mesh primitives in addition to writing your own importer. Our upcoming v4.0.0 release (coming out in a few months) will set you up for creating triangle primitives, but you'd still have a lot of work to do to write your own importer.

**Reply by Zi7ar21** [reply]
Will 4.0 cover Bounding Volume Hierarchy? Ray-Tracing a mesh without BVH is horribly slow, even when using a GPU...

**Reply by brigidatm** [reply]
there is another way to import scenes into the project and where can i find them? there are simple objects already implemented?

**Reply by hollasch** [reply]
The intent of the books is to teach you how to write your own ray tracer. If you want to extend this ray tracer to import OBJ files, then you will need to write the importer code, just as you did for the ray tracer. If you want primitives other than AARECT and spheres, you will need to implement those yourself as well.

**Reply by hollasch** [reply]
@Zi7ar21 — we are not introducing a new BVH other than the BVH already presented in v3.
